packages/schemas/src/types/hook.ts

```diff
@@ -13,3 +21,21 @@ export type HookEventPayload = {
   user?: Pick<User, (typeof userInfoSelectFields)[number]>;
   application?: Pick<Application, 'id' | 'type' | 'name' | 'description'>;
 } & Record<string, unknown>;
+
+const nonemptyUniqueHookEventsGuard = hookEventsGuard
+  .nonempty()
+  .transform((events) => deduplicate(events));
+
+export const createHookGuard = object({
+  // Note: ensure the user will not create a hook with an empty name.
+  name: string().min(1).optional(),
+  event: hookEventGuard.optional(),
+  events: nonemptyUniqueHookEventsGuard.optional(),
+  config: hookConfigGuard,
+  enabled: boolean().optional(),
+});
+
+export const updateHookGuard = createHookGuard
+  .omit({ events: true })
+  .deepPartial()
+  .extend({ events: nonemptyUniqueHookEventsGuard.optional() });
```

Comment:
it's find to put this at api level? didn't see the benefit here.

Reply:
Yeah, they're only used at api level. updated.